src/lib.rs

```diff
@@ -114,31 +121,61 @@ impl ToPromiseString for [Promise] {
     }
 }
 
-#[cfg(any(target_os = "bitrig", target_os = "openbsd"))]
+#[cfg(target_os = "openbsd")]
 mod openbsd;
 
-#[cfg(any(target_os = "bitrig", target_os = "openbsd"))]
-pub use openbsd::pledge_with_paths;
+#[cfg(target_os = "openbsd")]
+pub use openbsd::pledge;
 
-#[cfg(not(any(target_os = "bitrig", target_os = "openbsd")))]
-pub fn pledge_with_paths(_: &str, _: &[&std::path::Path]) -> Result<(), Error> {
+#[cfg(not(target_os = "openbsd"))]
+pub fn pledge<'p, 'e, P, E>(_: P, _: E) -> Result<(), Error>
+where P: Into<Option<&'p str>>, E: Into<Option<&'e str>> {
     Err(Error::UnsupportedPlatform)
 }
 
-pub fn pledge(promises: &str) -> Result<(), Error> {
-    pledge_with_paths(promises, &[])
+#[macro_export]
+macro_rules! pledge {
+    [$($promises:ident)*, $($execpromises:ident)*] => {
+        {
+            let mut promises = Vec::new();
+            let mut execpromises = Vec::new();
+            $(
+                promises.push(Promise::$promises);
+            )*
+            $(
+                execpromises.push(Promise::$execpromises);
+            )*
+            let promises = promises.to_promise_string();
+            let execpromises = execpromises.to_promise_string();
+            pledge(&*promises, &*execpromises)
+        }
+    };
 }
 
 #[macro_export]
-macro_rules! pledge {
-    ( $( $x:ident ),* ) => {
+macro_rules! pledge_promises {
+    [$($promises:ident)*] => {
         {
             let mut promises = Vec::new();
             $(
-                promises.push(Promise::$x);
+                promises.push(Promise::$promises);
             )*
-            let promises_str = promises.to_promise_string();
-            pledge(&promises_str)
+            let promises = promises.to_promise_string();
+            pledge(&*promises, None)
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! pledge_execpromises {
+    [$($execpromises:ident)*] => {
+        {
+            let mut execpromises = Vec::new();
+            $(
+                execpromises.push(Promise::$execpromises);
+            )*
+            let execpromises = execpromises.to_promise_string();
+            pledge(None, &*execpromises)
```

Comment:
strictly speaking `pledge_promises![p q r]` and `pledge_execpromises![e f g]` can be expressed as `pledge![p q r, ]` and `pledge![, e f g]` but I think a bit of syntactic sugar would be nice

Reply:
Yup, makes sense!